chia/consensus/block_creation.py

```diff
@@ -254,7 +254,7 @@ def create_foliage(
         )
         assert foliage_transaction_block is not None
 
-        foliage_transaction_block_hash: Optional[bytes32] = foliage_transaction_block.get_hash()
+        foliage_transaction_block_hash = foliage_transaction_block.get_hash()
```

Comment:
For the record, already discussed in a call, I don't personally bother overconstraining/indicating on every variable, but it's certainly a thing that can be done and has the upside of making complaints from mypy a bit more directly where you want them, depending.  Mypy will check from the `.get_hash()` return hint on to wherever `foliage_transaction_block_hash` gets used to make sure the types are consistent.  What is much more important is getting rid of `Any`s so that the hints actual indicate what is there rather than what you plan to treat it as.

```suggestion
        foliage_transaction_block_hash: bytes32 = foliage_transaction_block.get_hash()
```